lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/ts_computation_client.cpp

```diff
@@ -47,12 +48,16 @@ std::vector<ComputationClient::DataPtr> TSComputationClient::ExecuteComputation(
           torch_lazy_tensors::compiler::ts_backend::GenericComputationTS*>(
           computation.computation())
           ->graph();
-  torch::jit::Code function(graph, "");
-  torch::jit::InterpreterState interp(function);
+  torch::jit::GraphExecutor interp(graph, "");
   std::vector<torch::jit::IValue> stack;
   for (auto argument : arguments) {
     const auto ts_data =
         std::static_pointer_cast<TSComputationClient::TSData>(argument);
+    if (lazy_tensors::NNCComputationClient::HardwareDeviceType() == at::kCUDA &&
+        ts_data->data_.device().type() != at::kCUDA) {
+      LTC_CHECK_EQ(ts_data->data_.dim(), 0);
+      ts_data->data_ = ts_data->data_.cuda();
```

Comment:
When does this happen?

Do we want to implicitly convert arguments to cuda like this, vs. raising an error? This might not be the right comparison to make, but I'm thinking about if you have an `nn.module` that you put on a cuda device, and you pass cpu tensors as arguments to it; I think we raise an error in that context, rather than implicitly moving data between devices.

Reply:
Good point, this should never happen, I've replaced it with a check.

I think you're right that we don't do implicit transfer between different CUDA devices, although I'm pretty sure we do from CPU to CUDA for a few operators (indexing comes to mind as an example). We want to (and do) preserve that behavior with lazy devices.